client/src/components/Ticketing/ticketingmanager/TicketTypes/SubscriptionTypes.tsx

```diff
@@ -0,0 +1,374 @@
+import {
+  DataGrid,
+  GridColumns,
+  GridEditInputCell,
+  GridEventListener,
+  GridPreProcessEditCellProps,
+  GridRenderEditCellParams,
+  GridRowEditStopReasons,
+  GridRowId,
+  GridRowModel,
+  GridRowModes,
+  GridRowModesModel,
+  GridValueFormatterParams,
+  useGridApiContext,
+} from '@mui/x-data-grid';
+import {useFetchToken} from '../Event/components/ShowingUtils';
+import {Switch, Tooltip} from '@mui/material';
+import {FormButton} from '../Event/components/FormButton';
+import {FormDeleteButton} from '../Event/components/FormDeleteButton';
+import {formatUSD} from '../RefundOrders/RefundOrders';
+import {
+  apiCall,
+  getDefaultSubscriptionType,
+  useFetchSubscriptionTypes,
+  usePopUp,
+  validateSubscriptionType,
+} from './SubscriptionTypeUtils';
+import {
+  BackIcon,
+  CirclePlusIcon,
+  EditIcon,
+  SaveIcon,
+  TrashCanIcon,
+} from '../../Icons';
+
+export const SubscriptionTypes = () => {
+  const {token} = useFetchToken();
+  const {popUpProps, setPopUpProps, showPopUp, setShowPopUp} = usePopUp();
+  const {subscriptionTypes = [], setSubscriptionTypes} =
+    useFetchSubscriptionTypes();
+  const [rowModesModel, setRowModesModel] = useState<GridRowModesModel>({});
+  const [editing, setEditing] = useState(false);
+
+  const handleRowModesModelChange = (newRowModesModel: GridRowModesModel) => {
+    setRowModesModel(newRowModesModel);
+  };
+
+  const handleRowEditStart: GridEventListener<'rowEditStart'> = (_, event) => {
+    if (editing) {
+      event.defaultMuiPrevented = true;
+    } else {
+      setEditing(true);
+    }
+  };
+
+  const handleDeleteClick = (id: GridRowId) => {
+    const {id: current, ...rest} = rowModesModel;
+    setPopUpProps(
+      'Confirm Delete',
+      `Press continue to proceed with deletion`,
+      false,
+      `delete-subscription-row-${id}`,
+      async () => {
+        try {
+          await apiCall(
+            'DELETE',
+            `${process.env.REACT_APP_API_2_URL}/subscription-types/${id}`,
+            undefined,
+            token,
+          );
+          setRowModesModel(rest);
+          setSubscriptionTypes((types) =>
+            types.filter((type) => type.id !== id),
+          );
+        } catch (error) {
+          setPopUpProps(
+            'Delete Failed',
+            error.json
+              ? (await error.json()).error
+              : 'Error deleting subscription type',
+            false,
+            'delete-failed',
+          );
+        }
+      },
+    );
+  };
+
+  const handleRowEditStop: GridEventListener<'rowEditStop'> = (
+    params,
+    event,
+  ) => {
+    const {
+      rowFocusOut,
+      shiftTabKeyDown,
+      escapeKeyDown,
+      enterKeyDown,
+      tabKeyDown,
+    } = GridRowEditStopReasons;
+    if (
+      params.reason === rowFocusOut ||
+      params.reason === shiftTabKeyDown ||
+      params.reason === escapeKeyDown ||
+      params.reason === enterKeyDown ||
+      params.reason === tabKeyDown
+    ) {
```

Comment:
is there any particular reason why these 4 button presses were specified? I believe it might end up hindering accessibility since you can't tab between fields and can no longer "enter" to save.

Reply:
I think it might end up being confusing to have an explicit save button while also saving when the user leaves focus. It might be worth picking a singular option.

Reply:
I might be wrong, but I think if only rowFocusOut is left there then it’ll only save on button press or “enter” click

Reply:
I updated the handler so that enterKeyDown will save  👍